org.aimodels.aiy.json

```diff
@@ -0,0 +1,45 @@
+{
+  "app-id": "org.aimodels.aiy",
+  "runtime": "org.freedesktop.Platform",
+  "runtime-version": "22.08",
+  "sdk": "org.freedesktop.Sdk",
+  "command": "aiy",
+  "finish-args": [
+    "--share=network"
+  ],
+  "modules": [
+    {
+      "name": "aiy",
+      "buildsystem": "simple",
+      "sources": [
+        {
+          "type": "git",
+          "url": "https://github.com/visioninit/aiy.git",
+          "branch": "main"
+        }
+      ],
+      "modules": [
+        "resources/build/flatpak/pypi-dependencies.json"
+      ],
+      "build-commands": [
+        "pip3 install pyinstaller",
+        "pyinstaller --onefile aiy.py --name aiy --distpath dist/.",
```

Comment:
Why did you use pyinstaller? You have Python included in the Runtime.

Reply:
There was some issue in the flatpak with pygments that I wasn't able to track down. 

```
Traceback (most recent call last):
  File "/app/share/aiy.py", line 8, in <module>
    from rich.markdown import Markdown
  File "/app/lib/python3.10/site-packages/rich/markdown.py", line 18, in <module>
    from .syntax import Syntax
  File "/app/lib/python3.10/site-packages/rich/syntax.py", line 22, in <module>
    from pygments.lexer import Lexer
ModuleNotFoundError: No module named 'pygments'
```

Other modules that were downloaded and installed weren't throwing errors. I tried overriding the python-pip-generator to also download pigments (in case there was a version or location conflict), but it didn't change anything. I reviewed these issues with another person who looked over my build files and config and told me I had set it up correctly. 

I also tried setting up a venv but ran into the 'can't use pip to download normally' so I tried to build the executable and it worked. 

I assumed that something about the way the base package handles pygments or some of it's libraries may have caused some conflict when I tried to use it, but I don't know. I didn't change anything about how the code is set up between when I got the error above and when I built it into a binary with pyinstaller.

Reply:
I don't see pygments in your [pypi-dependencies.json](https://github.com/visioninit/aiy/blob/21224c12e65ce6aec2b24f3a366dad19f0a50152/resources/build/flatpak/pypi-dependencies.json). It looks like you have just forget to add it.

Reply:
It's included in the flatpak base image for 22.08 and is a dependency of rich, so it's not necessary to add it specifically. I did try adding it at some points to double check. The python-pip-generator has a list of the python modules that are already installed on the base image that it prevents them from downloading again.

Reply:
It is included rune Sdk, but not the runtime. I will take a look at it tomorrow.

Reply:
Sorry for the confusion. I'm not so familiar with flatpak yet. This is what I was looking at 

https://github.com/visioninit/aiy/blob/main/resources/build/flatpak/flatpak-pip-generator#L329 

```
# Python3 packages that come as part of org.freedesktop.Sdk.
system_packages = ['cython', 'easy_install', 'mako', 'markdown', 'meson', 'pip', 'pygments', 'setuptools', 'six', 'wheel']
# system_packages = ['cython', 'easy_install', 'mako', 'markdown', 'meson', 'pip',  'six', 'wheel']
```

In the aiy.py file that had the error on line 8

```
#!/usr/bin/env python3
import argparse
import os
import platform
import sys

from rich import pager
from rich.console import Console
from rich.markdown import Markdown
from rich.prompt import Prompt

from resources import config
from resources.conduit import get_completion
```

What I noticed is that the default modules and ```import pager```, ```rich.cosole import Console``` did not throw the error. I thought if it were something wrong with the imports or includes, it would have probably failed once it got to rich import pager. 

The priority for myself on this isn't high as I was informed that there isn't a strong case for publishing a cli app to flatpak. I'm here now to understand and assist in resolving this question. Let me know if there is anything else I can provide. If you'd like commit messages and the commands to reproduce the error I found I can look back and see if I have that still or if I can provide an example.

Reply:
Try adding this:
```json
{
    "name": "python3-pygments",
    "buildsystem": "simple",
    "build-commands": [
        "pip3 install --verbose --exists-action=i --no-index --find-links=\"file://${PWD}\" --prefix=${FLATPAK_DEST} \"pygments\" --no-build-isolation"
    ],
    "sources": [
        {
            "type": "file",
            "url": "https://files.pythonhosted.org/packages/0b/42/d9d95cc461f098f204cd20c85642ae40fbff81f74c300341b8d0e0df14e0/Pygments-2.14.0-py3-none-any.whl",
            "sha256": "fa7bd7bd2771287c0de303af8bfdfc731f51bd2c6a47ab69d117138893b82717",
            "x-checker-data": {
                "type": "pypi",
                "name": "Pygments",
                "packagetype": "bdist_wheel"
            }
        }
    ]
```
}